volmigrate/backend/etcd2/etcd2.go

```diff
@@ -0,0 +1,112 @@
+package etcd2
+
+import (
+	"path"
+	"strconv"
+
+	"github.com/contiv/errored"
+	"github.com/contiv/volplugin/volmigrate/backend"
+	"github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+func New(prefix string, etcdHosts []string) *Engine {
+	etcdCfg := client.Config{
+		Endpoints: etcdHosts,
+	}
+
+	etcdClient, err := client.New(etcdCfg)
+	if err != nil {
+		log.Fatalf("Failed to create etcd client: %s", err)
+	}
+
+	return &Engine{
+		etcdClient: client.NewKeysAPI(etcdClient),
+		prefix:     prefix,
+	}
+}
+
+// Engine is a migration engine for an etcd v2 datastore.
+type Engine struct {
+	etcdClient client.KeysAPI
+	prefix     string
+}
+
+// CurrentSchemaVersion returns the version of the last migration which was successfuly run.
+func (e *Engine) CurrentSchemaVersion() int64 {
+	resp, err := e.etcdClient.Get(context.Background(), path.Join(e.prefix, backend.SchemaVersionKey), nil)
+	if err != nil {
+		// TODO: check that this is actually a Not Found
```

Comment:
Perfect, I assumed there was a way to do this so I didn't bother writing my own.  Thanks!